src/definitions.ts

```diff
@@ -164,24 +164,31 @@ export interface UserDependencies {
   service: ISingleUserService;
 }
 
+export interface UserSocialProvider {
+  facebook?: UserSocial;
+  github?: UserSocial;
+  twitter?: UserSocial;
+  instagram?: UserSocial;
+  google?: UserSocial;
+  linkedin?: UserSocial;
+}
+
 export interface UserSocial {
```

Comment:
Make _this_ the `UserSocialProvider`, because this interface represents the social provider.